WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -187,22 +253,81 @@ final class ProductDetailPreviewViewModel: ObservableObject {
     }
 }
 
+// MARK: - Undo edits
+//
+extension ProductDetailPreviewViewModel {
+    func undoEdits(in field: ProductDetailField) {
+        guard let generatedAIProduct else {
+            return
+        }
+
+        guard let option = options[safe: selectedOptionIndex] else {
+            return
+        }
+
+        switch field {
+        case .name:
+            options[selectedOptionIndex] = NameSummaryDescOption(name: generatedAIProduct.names[selectedOptionIndex],
+                                                                 shortDescription: productShortDescription,
+                                                                 description: productDescription)
+        case .shortDescription:
+
+            options[selectedOptionIndex] = NameSummaryDescOption(name: productName,
+                                                                 shortDescription: generatedAIProduct.shortDescriptions[selectedOptionIndex],
+                                                                 description: productDescription)
+        case .description:
+            options[selectedOptionIndex] = NameSummaryDescOption(name: productName,
+                                                                 shortDescription: productShortDescription,
+                                                                 description: generatedAIProduct.descriptions[selectedOptionIndex])
+        }
```

Comment:
WDYT about avoiding the duplicated logic:

```suggestion
        let name = updatedField != .name ? productName :
            generatedAIProduct.names[selectedOptionIndex]

        let shortDescription = updatedField != .shortDescription ? productShortDescription :
            generatedAIProduct.shortDescriptions[selectedOptionIndex]

        let description = updatedField != .description ? productDescription :
            generatedAIProduct.descriptions[selectedOptionIndex]

        options[selectedOptionIndex] = NameSummaryDescOption(name: name, shortDescription: shortDescription, description: description)
```

Reply:
Nice suggestion. Added in https://github.com/woocommerce/woocommerce-ios/pull/13281/commits/8dc33ca19785c0cdbf2426296fd30d0e3c505cfb